pex/common.py

```diff
@@ -156,6 +157,19 @@ def safe_rmtree(directory):
     shutil.rmtree(directory, True)
 
 
+def safe_sleep(seconds):
+  """Ensure that the thread sleeps at a minimum the requested seconds.
+
+  Until Python 3.5, there was no guarantee that time.sleep() would actually sleep the requested
+  time. See https://docs.python.org/3/library/time.html#time.sleep."""
+  if sys.version_info[0:2] >= (3, 5):
+    time.sleep(seconds)
+  else:
+    start_time = time.time()
+    while time.time() - start_time < seconds:
+      pass
```

Comment:
Eek! Busy looping is for a handful of ms or less. This should still use time.sleep in the loop, but probably sleep the current defecit, keeping track of the defecit.